lib/pbench/server/filetree.py

```diff
@@ -336,72 +336,72 @@ def unpack(self, incoming: Path, results: Path):
             the tarball.
         """
         unpacked = incoming / f"{self.name}.unpack"
-        unpacked.mkdir(parents=True)
+        # unpacked.mkdir(parents=True)
 
         try:
             process = subprocess.run(
                 [
                     "tar",
-                    "-xf",
+                    "-x",
                     "--no-same-owner",
                     "--delay-directory-restore",
                     "--force-local",
-                    self.tarball_path,
+                    "--file=" + str(self.tarball_path),
                 ],
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballUnpackError(self.name, str(exc)) from exc
+            raise TarballUnpackError(self.tarball_path, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballUnpackError(
-                    self.name,
+                    self.tarball_path,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
+        find_command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
         try:
             process = subprocess.run(
-                command.split(),
+                find_command.split(),
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballModeChangeError(self.name, str(exc)) from exc
+            raise TarballModeChangeError(self.tarball_path, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballModeChangeError(
-                    self.name,
+                    self.tarball_path,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        self.unpacked = incoming
+        self.unpacked = incoming / self.name
         unpack = unpacked / self.name
 
         try:
-            shutil.move(str(unpack), str(self.unpacked))
+            shutil.move(unpack, incoming)
         except Exception as exc:
-            raise TarballUnpackError(
-                self.name,
-                f"Error in moving unpacked tarball to Unpacked directory: {str(exc)}",
-            )
             try:
-                os.remove(self.unpacked / self.name)
+                shutil.rmtree(self.unpacked)
             except FileNotFoundError:
                 raise TarballUnpackError(
-                    self.name, "Error in removing partially copied tarball"
+                    self.tarball_path, "Error in removing partially copied Tarball"
                 )
```

Comment:
we are removing partially copied Tarball inside the incoming directory. Should we mention incoming here?

Reply:
My instinct is to provide to the developer as much of the context for the failure as practical.  In this case, I think that that is the argument to the function (e.g., `self.unpacked` or `upacked` or whatever it turns out to be) and the description of the failure (which I think is just the `str()` of the `FileNotFoundError`).

However, my instinct is also that `FileNotFoundError` is a rather unlikely result -- that would mean that whatever we did to _create_ this file/directory **worked** (or, at least, _reported_ success) and that something else snuck in and deleted it before we could.  I suspect that we're more likely to hit a permissions error or some other file system error here.

In any case, we probably don't want to restrict ourselves to catching only `FileNotFoundError` here (and letting all others propagate out) -- we probably want to catch `Exception` and report what we got:  the thing is, this `except` clause is catching an error which occurring during an attempt to recover from from _**another** error_, so we probably don't want to do very much with it, other than to note the result (in the log) for a developer.  (I.e., we couldn't move the unpacked temporary directory, and _then_ we couldn't remove it -- we want to report the first of those loudly and the second somewhat more quietly -- mostly, we don't want the second to interfere with recovering from the first.)

The reason for handling it specially (i.e., for specifically catching `FileNotFoundError`, perhaps separately from catching other stuff) is to allow us to report it better/differently...but, since we're not really reporting it differently (e.g., we're still raising `TarballUnpackError`, albeit with custom message text), we could almost get by with just a `pass` instead of a `raise`, since we're going to report `TarballUnpackError` at line 395, anyway.  We just need to make sure that the message text adequately represents the situation.

The error-within-an-error situation should probably be logged, and once that's done we should probably report the outer error as though the inner one had not occurred, since the outer one is likely to be more interesting (at least, it happened _first_...).

Reply:
Maybe I can just log in the inner error and raise the exception from outer Try Except.

Reply:
As I mentioned [below](https://github.com/distributed-system-analysis/pbench/pull/2965#discussion_r972244778), I think it would be better to have `rmtree()` ignore or handle the errors instead of using a `try` block.  (And, yes, if we handle the errors, we should log them and only raise an exception from the next-outer scope.)